pkg/apis/duck/v1alpha1/subscribable_types.go

```diff
@@ -53,9 +53,9 @@ type SubscriberSpec struct {
 	// +optional
 	Generation int64 `json:"generation,omitempty"`
 	// +optional
-	SubscriberURI string `json:"subscriberURI,omitempty"`
+	SubscriberURI *apis.URL `json:"subscriberURI,omitempty"`
```

Comment:
I think you can make this a plain `apis.URL` (which has a valid empty value), and then you won't have to handle two error cases (`nil` and `apis.URL{}`).

Reply:
(Here, and in all the other `_types.go` files.

Reply:
If something is optional, it should be indicated with the pointer value so this seems right. However, should this really be optional I think is the bigger question. I think it's optional right now to ensure the DeprecatedRef above works correctly at the same time as this.
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required
I don't think apis.URL has a nil value when empty, so I think we should not do this.

Once SubscriberURI becomes non-optional, changing it to a non-pointer is fine. If that's the plan, I suppose I'd be fine making this switch now if it's indeed going to be non-optional later.

Reply:
`apis.URL` already has a nil-value which is the struct with all-empty-fields, which renders to a string as `""`. Code already needs to handle this case, so handling both pointer-`nil` and exists-but-empty adds complexity to a number of the codepaths without seeming to add a lot of value.